components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIGatewayPublisherImpl.java

```diff
@@ -310,4 +310,26 @@ private void saveEndpointConfig(String gwHome, String content) {
             }
         }
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void changeAPIState(API api, String status) throws GatewayException {
+        if (gwHome == null) {
+            //create the message to be sent to the gateway. This contains the basic details of the API and target
+            //lifecycle state.
+            APIDTO gatewayDTO = new APIDTO(APIMgtConstants.GatewayEventTypes.API_STATE_CHANGE);
+            gatewayDTO.setLabels(api.getLabels());
+            APISummary apiSummary = new APISummary(api.getId());
+            apiSummary.setName(api.getName());
+            apiSummary.setVersion(api.getVersion());
+            apiSummary.setContext(api.getContext());
+            apiSummary.setLifeCycleState(status);
+            gatewayDTO.setApiSummary(apiSummary);
+            publishToPublisherTopic(gatewayDTO);
+        } else {
+            //TODO save to file system: need to consider editor mode scenario
```

Comment:
@dewmini : Noted. In which case do we have a Git Issue for this so that this is tracked? It would be better to mention the Git issue in the ToDo as well. IMO this should be a L1 in the Git issue so that we fix this before releasing the feature.